src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp

```diff
@@ -101,6 +118,134 @@ class PolicyManager final : public Singleton<PolicyManager>
             });
     }
 
+    void validateConfiguration()
+    {
+        for (auto it = m_configuration.begin(); it != m_configuration.end(); it++)
+        {
+            if (it.key().compare("vulnerabilty-detection"))
+            {
+                validateAndConfigureVulnerabilityDetection(*it);
+            }
+            else if (it.key().compare("indexer"))
+            {
+                validateAndConfigureIndexer(*it);
+            }
+            else
+            {
+                throw std::runtime_error("Invalid JSON object.");
+            }
+        }
+    }
+
+    /**
+     * @brief Validates and configures the vulnerability detection based on the provided JSON object.
+     *
+     * This function takes a JSON object as input, which is expected to contain configuration
+     * information for an vulnerability detection. It validates the JSON object to ensure it contains the
+     * required fields and has valid values.
+     * If the validation passes, no exception is throw.
+     *
+     * @param vdObj A constant reference to a JSON object representing the vulnerability detection's configuration.
+     *
+     * @note This function assumes that the provided JSON object follows a specific format.
+     * @note If validation fails, this function throw exceptions.
+     *
+     */
+
+    void validateAndConfigureVulnerabilityDetection(const nlohmann::json& vdObj)
+    {
+        bool isValidUrl;
+
+        if (!vdObj["vulnerability-detection"].contains("enabled"))
+        {
+            throw std::runtime_error("Missing enabled field.");
+        }
+
+        if (!vdObj["vulnerability-detection"].contains("index-status"))
+        {
+            throw std::runtime_error("Missing index-status field.");
+        }
+
+        if (vdObj["vulnerability-detection"].contains("feed-update-interval"))
+        {
+            m_feedUpdateTime = Utils::parseStrToTime(vdObj["vulnerability-detection"]["feed-update-interval"]);
+        }
+
+        if (vdObj["vulnerability-detection"].contains("offline-url"))
+        {
+            isValidUrl = Utils::startsWith(vdObj["vulnerability-detection"]["offline-url"], "file") ||
+                         Utils::startsWith(vdObj["offline-url"], "https") ||
```

Comment:
Shouldn't it be `Utils::startsWith(vdObj["vulnerability-detection"]["offline-url"], "https")` instead of `Utils::startsWith(vdObj["offline-url"], "https")` ?